File: utils/timer_utils/javascript/timer.js
class Timer {

  start() {
    this.startTime = process.hrtime.bigint();
  }

  stop(message) {
    const endTime = process.hrtime.bigint();
    const diff = (endTime - this.startTime) / 1000000n;
    console.log(`${message}: ${diff} ms`);
  }
}

export default Timer;

File: utils/timer_utils/java/util/Timer.java
package util;


public class Timer {

    private long startTime;

    public void start() {
        startTime = System.nanoTime();
    }

    public void stop(String message) {
        long endTime = System.nanoTime();
        long timeTaken = endTime - startTime;
        System.out.println(message + ": "+ (timeTaken / 1000000) + "ms");
    }
}
